### PR TITLE
AP_RangeFinder: add serial support for GY-US42v2 rangefinder

### PR DESCRIPTION
**Pull request by peterbarker**
@Hwurzburg

### CODE REVIEW

File: libraries/SITL/SIM_RF_GYUS42v2.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  Simulator for the GYUS42v2 Serial rangefinder
+*/
+
+#include "SIM_RF_GYUS42v2.h"
+
+#include <GCS_MAVLink/GCS.h>
```
Comment (khancyr) [inline libraries/SITL/SIM_RF_GYUS42v2.cpp:21]: you don't need this one normally

Comment (peterbarker) [reply libraries/SITL/SIM_RF_GYUS42v2.cpp]: Fixed.

**Comment by muramura** [comment]
I will be using this device in I2C.
I will create an I2C driver for this device.
I hope to be able to access the I2C device in SITL.
I find it very inconvenient to increase the UART, as it is only CAN.

**Comment by Hwurzburg** [comment]
it runs under the Maxbotix I2C driver perfectly when placed in I2C mode with solder bridge to "L"

**Comment by muramura** [comment]
@Hwurzburg san. Thank you for your comment.
I will PR add this device name (I2C) to the device type in the config parameter.

**Comment by Hwurzburg** [comment]
In light of recent conversations, I suggest that the metadata name be GY-US042-Serial since it only applies to that device's serial mode and not the others....will test as soon as it stops raining here...probably Thurs sometime...

**Comment by Hwurzburg** [comment]
tested....works fine...although it took me a little while to figure out that just setting serial port protocol =9 also needed baud=9600....

File: libraries/AP_RangeFinder/AP_RangeFinder_GYUS42v2.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    const uint8_t num_read = uart->read(&buffer[buffer_used], ARRAY_SIZE(buffer)-buffer_used);
+    buffer_used += num_read;
+
+    if (buffer_used == 0) {
+        return false;
+    }
+
+    if (buffer[0] != 0x5A &&
+        !find_signature_in_buffer(1)) {
```
Comment (rmackay9) [inline libraries/AP_RangeFinder/AP_RangeFinder_GYUS42v2.cpp:51]: It's not a blocker but an alternative implementation would have been to throw away bytes until we see the header byte.  The LeddarVu8 driver has this kind of state machine (so does the Benewake serial driver). https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_RangeFinder/AP_RangeFinder_LeddarVu8.cpp#L131

Comment (peterbarker) [reply libraries/AP_RangeFinder/AP_RangeFinder_GYUS42v2.cpp:51]: One advantage in doing it the way presented here is that if your checksum doesn't match then you can scan the buffer and potentially find a real packet starting in the middle of what you thought was your payload.

If you're running a state machine you usually don't do that - you just throw away what didn't match and start from "want header" again.

**Pull request merged**